jsonschema/cli.py

```diff
@@ -17,31 +110,39 @@ def _namedAnyWithDefault(name):
     return namedAny(name)
 
 
-def _json_file(path):
-    with open(path) as file:
-        return json.load(file)
-
-
 parser = argparse.ArgumentParser(
     description="JSON Schema Validation CLI",
 )
 parser.add_argument(
     "-i", "--instance",
     action="append",
     dest="instances",
-    type=_json_file,
+    type=str,
     help=(
-        "a path to a JSON instance (i.e. filename.json) "
-        "to validate (may be specified multiple times)"
+        "A path to a JSON instance (i.e. filename.json) "
+        "to validate (may be specified multiple times). "
+        "If this option is not used, an instance can be given through stdin. "
+        "If neither -i nor stdin is given, only the schema is checked."
     ),
 )
 parser.add_argument(
     "-F", "--error-format",
     default="{error.instance}: {error.message}\n",
     help=(
-        "the format to use for each error output message, specified in "
+        "The format to use for each error output message, specified in "
```

Comment:
Maybe it should be an error if this is used with pretty?

Reply:
In that case the only way is to remove the default from argparse and manually process the arg. I added a new test for that.